module/os/linux/zfs/zfs_vfsops.c

```diff
@@ -1555,14 +1555,18 @@ zfs_domount(struct super_block *sb, zfs_mnt_t *zm, int silent)
 	/* Allocate a root inode for the filesystem. */
 	error = zfs_root(zfsvfs, &root_inode);
 	if (error) {
+		VERIFY3P(zfsvfs->z_os, !=, NULL);
```

Comment:
These VERIFYs seem like overkill. We'd have already crashed in `dmu_objset_is_snapshot()` above if this was NULL.